cmd/tools/asup/asup.go

```diff
@@ -0,0 +1,294 @@
+
+	// info about ONTAP host and instances
+	msg.Target, msg.Nodes, msg.Volumes, msg.Svms = getInstanceInfo(collectors, status)
+
+	return msg, nil
+}
+
+func getInstanceInfo(collectors []collector.Collector, status *matrix.Matrix) (*targetInfo, *instanceInfo, *instanceInfo, *instanceInfo) {
+	target := new(targetInfo)
+	nodes := new(instanceInfo)
+	vols := new(instanceInfo)
+	svms := new(instanceInfo)
+
+	// get ping value from poller metadata
+	target.Ping, _ = status.LazyGetValueFloat64("ping", "host")
+
+	// scan collectors
+
+	for _, c := range collectors {
+
+		if c.GetName() == "Zapi" {
+
+			if c.GetObject() == "Node" {
+				md := c.GetMetadata()
+				nodes.Count, _ = md.LazyGetValueInt64("count", "instance")
+				nodes.DataPoints, _ = md.LazyGetValueInt64("count", "data")
+				nodes.PollTime, _ = md.LazyGetValueInt64("poll_time", "data")
+				nodes.ApiTime, _ = md.LazyGetValueInt64("api_time", "data")
+				nodes.ParseTime, _ = md.LazyGetValueInt64("parse_time", "data")
+				nodes.PluginTime, _ = md.LazyGetValueInt64("plugin_time", "data")
+
+				target.Version = c.GetHostVersion()
+				target.Model = c.GetHostModel()
+				target.Serial = c.GetHostUUID()
+
+			} else if c.GetObject() == "Volume" {
+				md := c.GetMetadata()
+				vols.Count, _ = md.LazyGetValueInt64("count", "instance")
+				vols.DataPoints, _ = md.LazyGetValueInt64("count", "data")
+				vols.PollTime, _ = md.LazyGetValueInt64("poll_time", "data")
+				vols.ApiTime, _ = md.LazyGetValueInt64("api_time", "data")
+				vols.ParseTime, _ = md.LazyGetValueInt64("parse_time", "data")
+				vols.PluginTime, _ = md.LazyGetValueInt64("plugin_time", "data")
+
+				//} else if c.GetObject() == "Svm" {
+				//	md := c.GetMetadata()
+				//	svms.Count, _ = md.LazyGetValueInt64("count", "instance")
+				//	svms.DataPoints, _ = md.LazyGetValueInt64("count", "data")
+				//	svms.PollTime, _ = md.LazyGetValueInt64("poll_time", "data")
+				//	svms.ApiTime, _ = md.LazyGetValueInt64("api_time", "data")
+				//	svms.ParseTime, _ = md.LazyGetValueInt64("parse_time", "data")
+				//	svms.PluginTime, _ = md.LazyGetValueInt64("plugin_time", "data")
+			}
+		} else if c.GetName() == "ZapiPerf" {
+
+			if c.GetObject() == "Svm" {
+				md := c.GetMetadata()
+				svms.Count, _ = md.LazyGetValueInt64("count", "instance")
+				svms.DataPoints, _ = md.LazyGetValueInt64("count", "data")
+				svms.PollTime, _ = md.LazyGetValueInt64("poll_time", "data")
+				svms.ApiTime, _ = md.LazyGetValueInt64("api_time", "data")
+				svms.ParseTime, _ = md.LazyGetValueInt64("parse_time", "data")
+				svms.PluginTime, _ = md.LazyGetValueInt64("plugin_time", "data")
+			}
+		}
+	}
+
+	return target, nodes, vols, svms
+}
+
+func getCPUInfo() (string, uint8) {
+
+	var (
+		arch, countString, line string
+		fields                  []string
+		count                   uint64
+		output                  []byte
+		err                     error
+	)
+
+	if output, err = exec.Command("lscpu").Output(); err == nil {
```

Comment:
better to consider using https://github.com/shirou/gopsutil

Reply:
+1 I mentioned the same for asup payload limit checks

Reply:
changed all exec command to use library